sr_unix/ydb_thread_is_main.c

```diff
@@ -0,0 +1,42 @@
+/****************************************************************
+ *								*
+ * Copyright (c) 2018 YottaDB LLC. and/or its subsidiaries.	*
+ * All rights reserved.						*
+ *								*
+ *	This source code contains the intellectual property	*
+ *	of its copyright holder(s), and is made available	*
+ *	under a license.  If you do not know the terms of	*
+ *	the license, please stop and do not read further.	*
+ *								*
+ ****************************************************************/
+
+#include "mdef.h"
+
+#include "gtmxc_types.h"
+#include "error.h"
+#include "send_msg.h"
+#include "libydberrors.h"
+
+/* Simple YottaDB wrapper for gtm_is_main_thread() */
+int ydb_thread_is_main(void)
+{
+	int		status;
+	boolean_t	error_encountered;
+	DCL_THREADGBL_ACCESS;
+
+	SETUP_THREADGBL_ACCESS;
+	if (process_exiting)
+	{	/* YDB runtime environment not setup/available, no driving of errors */
+		send_msg_csa(CSA_ARG(NULL) VARLSTCNT(1) ERR_CALLINAFTERXIT);
+		return YDB_ERR_CALLINAFTERXIT;
+	}
+	ESTABLISH_NORET(ydb_simpleapi_ch, error_encountered);
+	if (error_encountered)
+	{	/* Some error occurred - return the error code to the caller ($ZSTATUS is set) */
+		REVERT;
+		return -(TREF(ydb_error_code));
+	}
+	status = gtm_is_main_thread();
+	REVERT;
+	return status ? YDB_OK : -1;
```

Comment:
As noted above in ydb_file_is_identical(), the -1 return code needs to be less hardcoded - probably given a #define in libyottadb.h and that name needs to be documented.

Reply:
Like the previous comment, I am hoping you will own this issue.

Reply:
Yes